main.tf

```diff
@@ -1,16 +1,165 @@
-  version = "1.0.0"
+data "aws_route53_zone" "zone" {
+  count   = var.zone_id == null ? 0 : 1
+  zone_id = var.zone_id
+}
 
-  enforce_case = "UPPER"
-  names        = [var.name]
-  tags         = var.tags
+data "aws_secretsmanager_secret_version" "dbpassword" {
+  secret_id = module.bitbucketdb.password_secretsmanager_arn
 }
 
 locals {
-  # tflint-ignore: terraform_unused_declarations
-  name = module.tags.name
-  # tflint-ignore: terraform_unused_declarations
-  tags = module.tags.tags_no_name
+  domain_name   = try(trimsuffix(data.aws_route53_zone.zone[0].name, "."), var.domain_name)
+  bitbucket_url = "${var.dns_prefix}.${local.domain_name}"
+  db_name       = replace(local.short_name, "-", "")
+  db_vpc_id     = var.db_vpc_id == null ? var.vpc_id : var.db_vpc_id
+  region        = data.aws_region.current.name
+  short_name    = substr(var.name, 0, 32)
+
+  configure_script = templatefile("${path.module}/templates/configure.sh.tpl",
+    {
+      admin_password = var.admin_password
+      admin_email    = var.admin_email
+      base_hostname  = local.bitbucket_url
+      db_url         = "jdbc:postgresql://${module.bitbucketdb.instance_connection_info.endpoint}/postgres"
+      db_username    = module.bitbucketdb.instance_connection_info.username
+      db_password    = replace(data.aws_secretsmanager_secret_version.dbpassword.secret_string, "$", "\\$")
+      license_key    = var.license_key
+      mount_point    = "/opt/atlassian/data"
+      region         = local.region
+      site_name      = var.site_name
+      volume_id      = aws_ebs_volume.data.id
+    }
+  )
+
+  db_allowed_security_groups = concat(
+    [aws_security_group.this.id],
+    var.db_additional_security_groups
+  )
+}
+
+resource "aws_ebs_volume" "data" {
+  availability_zone = var.availability_zone
+  size              = var.data_volume_size
+
+  tags = merge(var.tags,
+    { VolumeKey = var.volume_key }
+  )
+}
+
+resource "aws_autoscaling_group" "this" {
+  name_prefix               = var.name
+  desired_capacity          = var.asg_desired_capacity
+  health_check_grace_period = 600
+  health_check_type         = "EC2"
+  force_delete              = false
+  launch_configuration      = aws_launch_configuration.this.name
+  load_balancers            = [aws_elb.this.id]
+  max_size                  = var.asg_max_size
+  min_size                  = var.asg_min_size
+  wait_for_capacity_timeout = "15m"
+  vpc_zone_identifier       = [var.asg_subnet]
+
+  tag {
+    key                 = "Name"
+    propagate_at_launch = true
+    value               = var.name
+  }
+
+  tag {
+    key                 = "VolumeKey"
+    propagate_at_launch = true
+    value               = var.volume_key
+  }
+
+  dynamic "tag" {
+    for_each = var.tags
+
+    content {
+      key                 = tag.key
+      propagate_at_launch = true
+      value               = tag.value
+    }
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
```

Comment:
I actually made the healthcheck on this EC2 instead of ELB. I'm not sure how well Bitbucket will handle being aggressively recreated yet, so I was gun-shy about making something as important as a code repo easy to kill/replace. 

I did do basic testing, and it seems to handle it. But I honestly debated not having this behind an ASG at all out of paranoia.